composer/src/lib.rs

```diff
@@ -327,6 +351,27 @@ impl Builder {
         }
     }
 
+    /// get the name of the experiment
+    pub fn get_name(&self) -> String {
+        self.name.clone()
+    }
+
+    /// configure the `Builder` using the `BuilderConfigure` trait
+    pub fn configure(
+        self,
+        cfg: impl BuilderConfigure,
+    ) -> Result<Builder, Box<dyn std::error::Error>> {
+        cfg.configure(self)
+    }
+
+    /// current container length
+    pub fn next_container_ip(&self) -> String {
```

Comment:
It may be worth considering a `dev-dependency` on something like [`iprange`](https://docs.rs/iprange/0.6.3/iprange/) for this eventually.

Reply:
Yeah it's a bit fudged atm, but it looks like we already use `ipnetwork` so I could probably just use: https://docs.rs/ipnetwork/0.17.0/ipnetwork/struct.Ipv4Network.html#method.nth

Reply:
I've pushed a improved the PR description (I hope)